README.md

```diff
@@ -138,6 +138,47 @@ Will result in:
 </div>
 ```
 
+## Custom widgets
+
+You can provide your own custom widgets to a uiSchema for the following json data types:
+
+- `string`
+- `number`
+- `integer`
+- `boolean`
+- `date-time`
+
+```js
+const schema = {
+  type: "string"
+};
+
+const uiSchema = {
+  widget: (props) => {
+    return (
+      <input type="text"
+        className="custom"
+        value={props.value}
+        defaultValue={props.defaultValue}
+        required={props.required}
+        onChange={(event) => props.onChange(event.target.value)} />
+    );
+  }
+};
+
+render(<Form schema={schema} uiSchema={uiSchema} />);
+```
+
+The following props are passed to the widget component:
+
+- `schema`: The JSONSchema subschema object for this field;
+- `value`: The current value for this field;
+- `defaultValue`: The default value for this field;
+- `required`: The required status of this field;
+- `onChange`: The value change event handler; call it with the new value everytime it changes;
+- `placeholder`: The placeholder value, if any;
+- `options`: The list of options for `enum` fields;
```

Comment:
Hmm, `enum` is not a jsonschema datatype, it's a property specifying the field restricts the possible values for its type. How would you alter the docs to convey that?